pkg/archive/archive.go

```diff
@@ -322,6 +334,31 @@ func FileInfoHeader(name string, fi os.FileInfo, link string) (*tar.Header, erro
 	return hdr, nil
 }
 
+// fillGo18FileTypeBits fills type bits which have been removed on Go 1.9 archive/tar
```

Comment:
Should this override be in a separate file, fenced by a build flag? (`// go1.8`)?